src/Core/src/Hosting/MauiServiceProvider.cs

```diff
@@ -33,6 +28,12 @@ public MauiServiceProvider(IMauiServiceCollection collection)
 				baseType = baseType.BaseType;
 			}
 
+			foreach (var interfac in serviceType.GetInterfaces())
```

Comment:
I swapped interfaces with base types. It seems like if there's something registered against a concrete base type that should win..  Though I could see cases where you want the interface to win so perhaps it needs to walk each base type and check the concrete type/interfaces there to pick the winner?